### PR TITLE
Use "unsafe-wasm-eval" instead of "unsafe-eval" in script-src CSP

### PR DESCRIPTION
**Pull request by prplecake**
> In the CSP we are replacing the recently added `unsafe-eval` with the more strict `unsafe-wasm-eval`, as `unsafe-eval` is and was not really needed. This was initially intended in https://github.com/mastodon/mastodon/pull/18817 but we couldn't get it working until now.

([Thanks](https://github.com/mastodon/mastodon/pull/20606/#issuecomment-1313210289) @felixdoerre)

### CODE REVIEW

**Comment by felixdoerre** [comment]
Thanks. I think you changed the wrong `unsafe-eval`. You want to change the one in `production` (line 39) and not the one for dev (line 34)

**Comment by Gargron** [comment]
Please explain what you are doing and why

**Comment by felixdoerre** [comment]
In the CSP we are replacing the recently added `unsafe-eval` with the more strict `unsafe-wasm-eval`, as `unsafe-eval` is and was not really needed. This was initially intended in #18817 but we couldn't get it working until now.

**Comment by prplecake** [comment]
> Thanks. I think you changed the wrong `unsafe-eval`. You want to change the one in `production` (line 39) and not the one for dev (line 34)

D'oh! Fixed in last push.

**Review by Gargron — APPROVED** [review]

**Comment by Gargron** [comment]
Wait, is it actually supposed to be double-quoted like that?

File: config/initializers/content_security_policy.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ def host_to_url(str)
     p.worker_src  :self, :blob, assets_host
   else
     p.connect_src :self, :data, :blob, assets_host, media_host, Rails.configuration.x.streaming_api_base_url
-    p.script_src  :self, assets_host, :unsafe_eval
+    p.script_src  :self, assets_host, "'unsafe-wasm-eval'"
```
Comment (Gargron) [inline config/initializers/content_security_policy.rb:39]: Wait, according to the issue discussion, it's `wasm-unsafe-eval`, not `unsafe-wasm-eval`. Which is it?

**Comment by prplecake** [comment]
> Wait, is it actually supposed to be double-quoted like that?

Yes those double quotes are intentional, though I couldn't tell you why. See the diffs in the issue, single quoting didn't work for me but double quoting did. 

And I'm pretty sure I have the correct directive, also see the issue. "unsafe-wasm-eval" doesn't produce CSP errors from the browser.

**Comment by Gargron** [comment]
There are no search results for "unsafe-wasm-eval" but there are Google/Firefox pages referencing "wasm-unsafe-eval", also the person who told you about the quotes used "wasm-unsafe-eval" in their code snippet.

**Comment by prplecake** [comment]
You're right, it should be `wasm-unsafe-eval`.

**Pull request merged**